### PR TITLE
feat: Support ALTER SET SYSTEM barrier_interval_ms

### PR DESCRIPTION
**Pull request by hzxa21**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Change `barrier_interval_ms` to be mutable with valid range [100ms, +inf)
- Barrier manager in meta will listen on `barrier_interval_ms` changes and adjust the barrier timer.
- Global memory manager will listen on `barrier_interval_ms` changes and adjust the tick interval.
- Source executor will refresh `max_wait_barrier_time_ms` when the stream is pasued.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/meta/src/barrier/mod.rs [hunk 1]
```diff
@@ -598,7 +598,21 @@ where
...
+                        if new_interval != min_interval.period() {
+                            min_interval = tokio::time::interval(new_interval);
+                            min_interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
+                        }
+                        self.scheduled_barriers
+                            .set_checkpoint_frequency(p.checkpoint_frequency() as usize)
+                    }
+
+                    // Handle other changes
+                    self.handle_local_notification(notification);
```
Comment (BugenZhao) [inline src/meta/src/barrier/mod.rs:615]: Seems we've already handled all possible changes?

**Review by BugenZhao — APPROVED** [review]
LGTM

File: src/common/src/system_param/mod.rs [hunk 2]
```diff
@@ -36,7 +36,7 @@ macro_rules! for_all_undeprecated_params {
         // Hack: match trailing fields to implement `for_all_params`
         $(, { $field:ident, $type:ty, $default:expr, $is_mutable:expr })*) => {
         $macro! {
-            { barrier_interval_ms, u32, Some(1000_u32), false },
+            { barrier_interval_ms, u32, Some(1000_u32), true },
```
Comment (BugenZhao) [inline src/common/src/system_param/mod.rs:39]: BTW, setting this value to an arbitrarily large number may cause OOM. Perhaps we should document a range of empirical values somewhere for reference.

**Comment by yuhao-su** [comment]
LGTM

File: src/meta/src/barrier/mod.rs (hunk 1 above)
Comment (hzxa21) [reply src/meta/src/barrier/mod.rs:615]: Right. Barrier manager only reacts to ckpt frequency and barrier interval changes. ckpt frequency change is handled in `handle_local_notification`.

Comment (BugenZhao) [reply src/meta/src/barrier/mod.rs:615]: > ckpt frequency change is handled in `handle_local_notification`.

Seems already handled in L610 🥵

Comment (hzxa21) [reply src/meta/src/barrier/mod.rs]: lol. You are right. Fixed.

File: src/common/src/system_param/mod.rs (hunk 2 above)
Comment (hzxa21) [reply src/common/src/system_param/mod.rs:39]: Added notice messages in the ALTER SYSTEM if barrier interval or checkpoint frequency is set too high.

**Review by BugenZhao — APPROVED** [review]
LGTM

**Pull request merged**